documentation/wiki/Mac-Debugging.md

```diff
@@ -0,0 +1,38 @@
+#Debugging with MacOS
+Say unit tests are failing from Microsoft.Build.Engine.UnitTests_netcoreapp2.1_x64
+* Open terminal 
+  * Hit command-space, type terminal, hit enter. Alternatively, hit magnifying glass (spotlight) in upper-right corner and search for terminal.)
+* Build something
+  * Navigation in terminal is similar to command prompt (cd), although you type `ls` in place of `dir`.
+  * **Use ./build.sh instead of .\build.cmd.**
+* Type `find . -name Microsoft.Build.Engine.UnitTests_netcoreapp2.1_x64.log`
+  * This should print out a path (from your current working directory) to the relevant log file.
+* Type `tail <path from previous step>`
+  * This prints out the last part of that file.
+  * You can also just open it normally from a finder window.
+  * This file contains the standard output from the last run.
+  * You may notice that the file ends with `=== COMMAND LINE ===` followed by a single (long) command line statement.
+* Copy the command line statement from the previous step. Remove the portion after the redirection (`>` character). You may notice that part redirects output to the file you’re viewing.
+  * The last part (`2>&1`) redirects standard error (using `2>`) to the same place as where standard out is going (`&1`), in this case this log file.
+* Append `./build.sh &&` to the beginning of the truncated command line statement.
+* Append `-method ` and the name of the method you want to test to the end.
```

Comment:
I wanted to say "link to the docs on how to do filters" but I can't find any such docs :(

Probably worth mentioning `-class` to run all tests in a class, and that the name provided must be fully qualified (including namespace).

Reply:
I tried ./build.sh -class..., and it told me that ./build.sh has no -class switch. Did you mean adding -class to something else?

Reply:
yes, to the test command line

Reply:
I'm not sure in what situations that would be helpful. If you know what class, then you presumably ran all tests at some point, and if you ran all tests, you can look at the html file and find which methods specifically, in which case you don't need to run a class full of tests. If you already know that a particular method failed, and you want to know if your fix fixed it, you'll want to test the method specifically, not the whole class. The log from stdout is a mess, and although it's possible, it's difficult to figure out what actually went wrong. stderr is better, but it's still less helpful than the html file because it doesn't really say why a test failed. In summary, what's the use case?

Reply:
I use `-class` probably 1/4 of the time that I run tests on the command line.

It's particularly useful when refactoring product code in a particular area: if it breaks something, it's likely to be in "the tests that cover this stuff", as opposed to entirely unrelated tests. But you don't know exactly _what_ might break.

It's also nice when you're changing a test utility method used in multiple test cases, or if multiple test cases failed for closely related reasons.

You certainly could list all of the failing tests individually, but I generally feel that if you're going to run more than one test, it's not worth scoping it to the minimum subset of interest.